### DISCUSSION TITLE
Alternatives to Ethers Byteslike or JsonFragment types

### DISCUSSION THREAD

**Discussion by FuzzB0t** [wevm/viem#1150] [Question]
Are there any types like Byteslike or JsonFragment as there are in the ethers.utils or abi libraries?

**Comment by jxom** [comment]
Byteslike = `Hex | ByteArray`

We don’t have concept of “fragments” as they’re merely a redundant pattern - viem has “abi items”. What are you using a JsonFragment for? I can tell you the alternative for that. If it’s for parsing/formatting an abi item, you can use: `parseAbiItem`/`formatAbiItem`

**Reply by FuzzB0t** [reply]
I'm enforcing the type on a parameter expected to be a JsonFragment

**Reply by FuzzB0t** [reply]
Im guessing I can replace that with Abi Item? Although JSON Fragments can include functions as well so maybe a union of both?
